actions/campaigns.go

```diff
@@ -232,11 +227,23 @@ func PostCampaign(c *gin.Context) {
 		return
 	}
 
+	template, err := storage.GetTemplateByName(c, body.Name)
+	if err != nil {
+		c.JSON(http.StatusUnprocessableEntity, gin.H{
+			"message": "Template with that name does not exists",
+		})
+		return
+	}
+
 	campaign := &entities.Campaign{
-		Name:         body.Name,
-		UserID:       user.ID,
-		TemplateName: body.TemplateName,
-		Status:       entities.StatusDraft,
+		Name:   body.Name,
+		UserID: user.ID,
+		Template: entities.Template{
```

Comment:
This doesn't need to be a new template object, you can use the `template` variable here.